### PR TITLE
Chore/upgrade pupil spa to angular 8

### PR DESCRIPTION
**Pull request by jon-shipley**
Upgrade pupil app to angular 8

### CODE REVIEW

File: pupil-spa/src/app/services/config/config.service.ts [hunk 1]
```diff
@@ -86,22 +86,29 @@ export function loadConfigMockService(): Function {
...
-        APP_CONFIG = Object.assign(t, envResponse);
-        resolve(true);
-      });
+    return new Promise(async (resolve, reject) => {
+      await this.http.get('/public/config.json', { observe: 'response' })
+        .pipe(first())
+        .toPromise()
+        .then(data => {
+          const t = new AppConfig();
+          console.log('data ', data)
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/app/services/config/config.service.ts]: can we get rid of the log?

File: pupil-spa/src/polyfills.ts [hunk 2]
```diff
@@ -11,35 +11,19 @@
...
+ * Learn more in https://angular.io/guide/browser-support
  */
 
 /***************************************************************************************************
- * BROWSER POLYFILLS
- */
+* BROWSER POLYFILLS
+*/
 
-/** IE9, IE10 and IE11 requires all of the following polyfills. **/
```
Comment (GeorgeChatzigiannis) [inline pupil-spa/src/polyfills.ts]: Are these polyfill imports:
a) not used on our codebase
b) not required as angular 8 is exposing these in a different way
c)any other reason that is not obvious to me

File: pupil-spa/src/app/services/config/config.service.ts (hunk 1 above)
Comment (jon-shipley) [reply pupil-spa/src/app/services/config/config.service.ts]: Yes, good spot.

File: pupil-spa/src/polyfills.ts (hunk 2 above)
Comment (jon-shipley) [reply pupil-spa/src/polyfills.ts]: Angular 8 uses differential loading - see the `browserlist` file to see what browsers we are targetting.  The polyfills will be created to support these browsers as required.

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/polyfills.ts]: ok good to know.

**Comment by GuyHarwood** [comment]
yes, its a good point @GeorgeChatzigiannis 
will need thorough testing across the browser space

**Pull request merged**